src/yardstick/store/labels.py

```diff
@@ -22,33 +22,52 @@ def label_store_root(store_root: str = None) -> str:
 
 
 def store_filename_by_entry(entry: artifact.LabelEntry) -> str:
-    if not entry.source or entry.source == label.MANUAL_SOURCE:
-        if entry.image.exact:
-            return f"{naming.image.encode(entry.image.exact)}/{entry.ID}.json"
-        return f"{entry.ID}.json"
     if entry.image.exact:
-        return f"{naming.image.encode(entry.image.exact)}/{entry.source}.json"
-    return f"{entry.source}.json"
+        return f"{naming.image.encode(entry.image.exact)}/{entry.ID}.json"
+    return f"{entry.ID}.json"
 
 
 def store_path(filename: str, store_root: str = None) -> str:
     return os.path.join(label_store_root(store_root=store_root), filename)
 
 
 def append_and_update(
-    new_and_modified_entries: List[artifact.LabelEntry], delete_entries: List[str] = None, store_root: str = None
+    new_and_modified_entries: List[artifact.LabelEntry], delete_entries: List[artifact.LabelEntry] = None, store_root: str = None
 ) -> List[artifact.LabelEntry]:
     for label_entry in delete_entries:
         filepath = store_path(filename=store_filename_by_entry(entry=label_entry))
         try:
+            logging.debug(f"deleting label {label_entry.ID} from {filepath}")
             os.remove(filepath)
-        except:  # pylint: disable=bare-except
-            pass
-
-    overwrite_all(label_entries=new_and_modified_entries, store_root=store_root)
+        except FileNotFoundError:
+            logging.debug(f"skipping deleting on {label_entry.ID} from {filepath}: File not found")
+        except Exception as e:  # pylint: disable=broad-except
+            logging.error(f"failed to delete label {label_entry.ID} from {filepath}: {e}")
+            raise e
+
+    save_label_entries(label_entries=new_and_modified_entries, store_root=store_root)
+
+
+def delete_labels_by_id(label_ids_to_delete: List[str], store_root: str = None) -> List[str]:
+    """delete_entries takes a list of ids to be deleted and returns a list of deleted files.
+    FileNotFound exceptions are ignored."""
+    label_store_dir = label_store_root(store_root=store_root)
+    deleted_ids = []
+    for label_id in label_ids_to_delete:
+        g = f"{label_store_dir}/**/{label_id}.json"
+        for p in glob.glob(g):
+            try:
+                os.remove(p)
+                deleted_ids.append(label_id)
+            except FileNotFoundError:
+                logging.debug(f"skipping deleting on {label_id} from {p}: File not found")
+            except Exception as e:  # pylint: disable=broad-except
+                logging.error(f"failed to delete label {label_id} from {p}: {e}")
+                raise e
+    return deleted_ids
 
 
-def overwrite_all(label_entries: List[artifact.LabelEntry], store_root: str = None):
+def save_label_entries(label_entries: List[artifact.LabelEntry], store_root: str = None):
```

Comment:
nit: naming this as `save` would be more consistent with the other stores and would encourage use from the module directly instead of importing a single function:
```python 
# ok
from yardstick.store.labels import save_label_entries
save_label_entries(items)

# better
from yardstick import store
store.labels.save(items)
```

Reply:
That makes sense. I'll probably change the imports too. There are a lot of places where it's imported like this `from yardstick.store.labels import delete_labels_by_id`, in which case having the full name really helps readability.